### PR TITLE
Unified Window

### LINKED ISSUE #81: "Recent buttercups" shows deleted archives
1. Create an archive
2. Delete the file
3. It still shows in the "Recent buttercups" list

### LINKED ISSUE #293: Open last group on archive unlock
Currently the last group open in an archive seems to be the last created(?).. It would be best if each group selection stored the following in localstorage (as a suggestion): `lastGroup:(archiveID)/(groupID)`. When an archive is opened this value could be checked and, if it's valid, could be selected (only of the group exists).

### LINKED ISSUE #302: Unified archive windows
Right now each archive opens in a new window, but that doesn't fit the usage and workflow of Buttercup products. Archives are usually "added" to workspace and not "opened" like a file.

The main window should be mix of all archives directly (after you unlock each) and you should be able to search and do other operations across your archives.

### LINKED ISSUE #308: TypeError on first time open
I downloaded the `dmg` file, and when opened it, I got this pop-up error:
```
TypeError: this._buildProcedures[type] is not a function
    at WindowManager.buildWindowOfType (/Applications/Buttercup.app/Contents/Resources/app.asar/dist/app.js:3411:47)
    at App.module.exports.__WEBPACK_IMPORTED_MODULE_1_electron__.app.on (/Applications/Buttercup.app/Contents/Resources/app.asar/dist/app.js:33402:19)
```

The app opened after pressing `OK`

### LINKED ISSUE #309: Dragging file over app in the background triggers error
Hi!

I noticed that when I drag a file that is not readable by the app (for example .txt's) over Buttercup - even when it's just sitting in the background, a drag and drop error message pops up. Even though this is not the biggest of deals, it get's quite annoying over time (the obvious solution is to hide it at all times). 

error-message

dragging-error

Buttercup 0.18.0
macOS Sierra 10.12.5

### LINKED ISSUE #310: Got this error message at first run
Hey guys ...
first of all , thanks for your great work
but when i open buttercup for first time , i got this message , can you explain what is going on ?
screen shot 2017-07-06 at 7 52 20 pm

### LINKED ISSUE #319: Import incorrectly tied to file association
In the [import code](https://github.com/buttercup/buttercup-desktop/blob/af1a33470678d359a885363c70fbf62ce166152a/src/main/lib/buttercup.js#L17) it seems that `'csv'` is tied to lastpass, when it should be generic (other managers will export CSV).

This is mentioned in buttercup/buttercup-importer#17.

### LINKED ISSUE #322: [Linux] desktop file is lacking a Categories= entry
The desktop file is lacking a `Categories=` entry in the `*.desktop` file.

According to the [menu spec](https://standards.freedesktop.org/menu-spec/latest/apa.html),

> By including one of the Main Categories in an application's desktop entry file, the application will be ensured that it will show up in a section of the application menu dedicated to this category. If multiple Main Categories are included in a single desktop entry file, the entry may appear more than once in the menu. 

Hence, please add at least one of the following in the `Categories=` key.

Main Category | Description | Notes
-- | -- | --
AudioVideo | Application for presenting, creating, or processing multimedia (audio/video) |  
Audio | An audio application | Desktop entry must include AudioVideo as well
Video | A video application | Desktop entry must include AudioVideo as well
Development | An application for development |  
Education | Educational software |  
Game | A game |  
Graphics | Application for viewing, creating, or processing graphics |  
Network | Network application such as a web browser |  
Office | An office type application |  
Science | Scientific software |  
Settings | Settings applications | Entries may appear in a separate menu or as part of a 		"Control Center"
System | System application, "System Tools" such as say a log viewer or network monitor |  
Utility | Small utility application, "Accessories"

In addition, you could specify one or more from the longer list of [Additional Categories](https://standards.freedesktop.org/menu-spec/latest/apas02.html).

Please test the result with `desktop-file-validate` and make sure it passes.

electron-builder offers native support for this since [v19.22.1](https://github.com/electron-userland/electron-builder/releases/tag/v19.22.1).

Reference: https://github.com/AppImage/AppImageHub/issues/2

### PR DESCRIPTION
**Pull request by sallar**
image

image

Fixes #319 
Fixes #81 
Fixes #310 
Fixes #293 
Fixes #302 
Fixes #308 
Fixes #309 
Fixes #322

### CODE REVIEW

File: app/package.json [hunk 1]
```diff
@@ -1,13 +1,14 @@
 {
   "name": "buttercup",
-  "version": "0.18.2",
+  "version": "0.19.0",
```
Comment (perry-mitchell) [inline app/package.json]: 😍

File: package.json [hunk 2]
```diff
@@ -126,7 +126,7 @@
     "babel-register": "^6.18.0",
     "babili-webpack-plugin": "^0.1.1",
     "buttercup-importer": "~0.9.2",
-    "buttercup-web": "~0.32.0",
+    "buttercup-web": "^0.32.1",
```
Comment (perry-mitchell) [inline package.json]: Probably best to keep this locked to patches.

File: src/renderer/index.js [hunk 3]
```diff
@@ -1,58 +1,58 @@
...
+} from '../shared/actions/archives';
+import { showHistoryPasswordPrompt } from '../shared/buttercup/import';
 import { setupShortcuts } from './system/shortcuts';
 import Root from './containers/root';
 
 // Unhandled rejections
-const unhandled = require('electron-unhandled');
-unhandled();
+// const unhandled = require('electron-unhandled');
+// unhandled();
```
Comment (perry-mitchell) [inline src/renderer/index.js]: Should we bin this? Or was this kind of a pending item..?

Comment (sallar) [inline src/renderer/index.js]: I commented this out temporarily. uncommenting again :)

File: package.json (hunk 2 above)
Comment (sallar) [inline package.json]: Ah sure this was a merge conflict

File: app/package.json (hunk 1 above)
Comment (sallar) [inline app/package.json]: 👍 😱

**Comment by perry-mitchell** [comment]
@sallar I was thinking that we might consider making this beta already.. and perhaps leap to version `0.20.0` to further that. What are your thoughts?

**Pull request merged**